actors/miner/tests/util.rs

```diff
@@ -806,6 +843,148 @@ impl ActorHarness {
         Ok(self.get_sector(rt, sector_number))
     }
 
+    pub fn prove_commit_sectors_ni(
+        &self,
+        rt: &MockRuntime,
+        mut params: ProveCommitSectorsNIParams,
+        first_for_miner: bool,
+        mut fail_fn: impl FnMut(&mut SectorNIActivationInfo, usize) -> bool,
+    ) -> Result<ProveCommitSectorsNIReturn, ActorError> {
+        let failed_sectors = params.sectors.iter_mut().enumerate().fold(
+            HashSet::new(),
+            |mut failed_sectors, (index, sector)| {
+                if fail_fn(sector, index) {
+                    failed_sectors.insert(sector.sealing_number);
+                }
+
+                failed_sectors
+            },
+        );
+        let fail_count = failed_sectors.len();
+
+        let expected_success_sectors = params
+            .sectors
+            .iter()
+            .map(|s| s.sealing_number)
+            .filter(|s| !failed_sectors.contains(&s))
+            .collect::<Vec<_>>();
+        let expected_success_count = expected_success_sectors.len();
+        assert_eq!(params.sectors.len(), expected_success_count + fail_count);
+
+        rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, self.worker);
+
+        let randomness = Randomness(TEST_RANDOMNESS_ARRAY_FROM_ONE.to_vec());
+
+        let entropy = serialize(&rt.receiver, "address for get verify info").unwrap();
+
+        params.sectors.iter().for_each(|s| {
+            rt.expect_get_randomness_from_tickets(
+                fil_actors_runtime::runtime::DomainSeparationTag::SealRandomness,
+                s.seal_rand_epoch,
+                entropy.to_vec(),
+                TEST_RANDOMNESS_ARRAY_FROM_ONE,
+            );
+        });
+
+        params.sectors.iter().filter(|s| !failed_sectors.contains(&s.sector_number)).for_each(
+            |s| {
+                expect_sector_event(rt, "sector-activated", &s.sealing_number, None, &vec![]);
+            },
+        );
+
+        let seal_verify_info = params
+            .sectors
+            .iter()
+            .map(|sector| AggregateSealVerifyInfo {
+                sector_number: sector.sealing_number,
+                randomness: randomness.clone(),
+                interactive_randomness: Randomness(vec![1u8; 32]),
+                sealed_cid: sector.sealed_cid.clone(),
+                unsealed_cid: CompactCommD::empty().get_cid(params.seal_proof_type).unwrap(),
+            })
+            .collect::<Vec<_>>();
+
+        rt.expect_aggregate_verify_seals(seal_verify_info, params.aggregate_proof.to_vec(), Ok(()));
+        rt.expect_validate_caller_addr(self.caller_addrs());
+
+        self.expect_query_network_info(rt);
+
+        if params.sectors.len() > 5 {
+            let aggregate_fee = aggregate_prove_commit_network_fee(
+                params.sectors.len() - fail_count - 5,
```

Comment:
Either the test at line 912 should be `params.sectors.len() - fail_count > 5` or the fail count should not be subtracted here. As-is, this can give a negative (underflow) sector count, or expect the wrong fee. You should be able to hit this with a test (submit 6, fail 2), and then fix the expectation.